components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/internal/util/WSO2PayloadUtils.java

```diff
@@ -325,4 +334,40 @@ private static boolean isCredentialUpdateFlow(String eventName) {
 
         return IdentityEventConstants.Event.POST_UPDATE_CREDENTIAL_BY_SCIM.equals(eventName);
     }
+
+    public static UserClaim generateUserClaim(String claimKey, String claimValue, String tenantDomain) {
+
+        UserClaim.Builder userClaimBuilder = new UserClaim.Builder()
+                .uri(claimKey);
+
+        String multiAttributeSeparator = FrameworkUtils.getMultiAttributeSeparator();
+        if (isMultiValuedClaim(claimKey, tenantDomain)) {
+            userClaimBuilder.value(StringUtils.isBlank(claimValue) ? new String[]{} :
+                    claimValue.split(Pattern.quote(multiAttributeSeparator)));
+        } else {
+            userClaimBuilder.value(claimValue);
+        }
+
+        return userClaimBuilder.build();
+    }
+
+    private static boolean isMultiValuedClaim(String claimUri, String tenantDomain) {
+
+        ClaimMetadataManagementService claimMetadataManagementService =
+                WSO2EventHookHandlerDataHolder.getInstance().getClaimMetadataManagementService();
+
+        try {
+            Optional<LocalClaim>
+                    localClaim = claimMetadataManagementService.getLocalClaim(claimUri, tenantDomain);
+
+            if (localClaim.isPresent()) {
+                return Boolean.parseBoolean(localClaim.get().getClaimProperty(ClaimConstants.MULTI_VALUED_PROPERTY));
+            }
```

Comment:
Shall we add a debug log when `localClaim.isPresent() == false` saying that the "Claim not found for claim URI: " + claimUri ?